sound/soc/sof/ipc4-loader.c

```diff
@@ -270,6 +398,32 @@ int sof_ipc4_query_fw_configuration(struct snd_sof_dev *sdev)
 	return ret;
 }
 
+int sof_ipc4_reload_fw_libraries(struct snd_sof_dev *sdev)
+{
+	struct sof_ipc4_fw_data *ipc4_data = sdev->private;
+	struct sof_ipc4_fw_library *fw_lib;
+	unsigned long lib_id;
+	int ret = 0;
+
+	/*
+	 * It is not an error if the callback is not set at this point since the
+	 * loaded topology did not needed external libraries to be loaded
```

Comment:
typo: did not need

Reply:
not sure I follow this comment @ujfalusi. Why is not an error "at this point"?

Reply:
We call this function unconditionally on non first boot cases. If the ipc4_data->load_library is not set and we don't have external libraries then it is just fine to return with 0.
If we would have needed external libraries and the ipc4_data->load_library is not set then the tplg parsing would have failed.

I don't think the upper layer should know much about libraries, number of libraries, etc.

The check is for sanity and might be dropped since the xa_for_each_start() would have 0 run in this case...

Reply:
I think it is safe to drop this check and the comment (along with the typo)